notification-server/src/bin/service.rs

```diff
@@ -112,28 +127,43 @@ async fn main() -> anyhow::Result<()> {
     let database_connection = DatabaseConnection::create(args.db_connection).await?;
 
     let mut concordium_client = Client::new(endpoint).await?;
-    let mut receiver = concordium_client.get_finalized_blocks().await?;
-    while let Some(v) = receiver.next().await {
-        let block_hash = v?.block_hash;
-        println!("Blockhash: {:?}", block_hash);
-        let transactions = concordium_client
-            .get_block_transaction_events(block_hash)
-            .await?
-            .response;
-        for result in process(transactions).await.iter() {
-            println!("address: {}, amount: {}", result.address, result.amount);
-            for device in database_connection
-                .prepared
-                .get_devices_from_account(result.address)
+    loop {
```

Comment:
Why do we have an outer `loop` and an inner `while` loop. The outer `loop` seems redundant?

Reply:
The outer loop is to catch connection error while issues occurs on the stream. Basically just a retry.

The inner traverses the chain. 

I will split the code up to show this more clearly